sphinx/ext/autodoc/importer.py

```diff
@@ -20,16 +21,86 @@
     isclass,
     isenumclass,
     safe_getattr,
+    unwrap_all,
 )
 
 if TYPE_CHECKING:
+    from collections.abc import Callable, Generator, Mapping
     from types import ModuleType
+    from typing import Any
 
     from sphinx.ext.autodoc import ObjectMember
 
 logger = logging.getLogger(__name__)
 
 
+def _filter_enum_dict(
+    enum_class: type[Enum],
+    attrgetter: Callable[[Any, str, Any], Any],
+    enum_class_dict: Mapping[str, object],
+) -> Generator[tuple[str, type, Any], None, None]:
```

Comment:
Can you just add a doc string, briefly explaining the function and what it yields

Reply:
Sure,. I'll do it in a few moments